xbmc/addons/interfaces/GUI/ListItem.cpp

```diff
@@ -87,5 +112,236 @@ void Interface_GUIListItem::destroy(void* kodiBase, void* handle)
     delete item;
 }
 
+char* Interface_GUIListItem::get_label(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CFileItemPtr* item = static_cast<CFileItemPtr*>(handle);
+  if (!addon || !item)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, item, addon ? addon->ID().c_str() : "unknown");
+    return nullptr;
+  }
+
+  if (item->get() == nullptr)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIListItem::%s - empty list item called on addon '%s'", __FUNCTION__, addon->ID().c_str());
+    return nullptr;
+  }
+
+  char* ret;
+  Interface_GUIGeneral::lock();
+  ret = strdup(item->get()->GetLabel().c_str());
```

Comment:
I would use <code>*item->Foo()</code> instead of <code>item->get()->Foo()</code> here and at several other places.